specs/electra/beacon-chain.md

```diff
@@ -789,6 +841,56 @@ def slash_validator(state: BeaconState,
 
 ## Beacon chain state transition function
 
+### Execution engine
+
+#### Request data
+
+#### Engine APIs
+
+##### Modified `notify_new_payload`
+
+*Note*: The function `notify_new_payload` is modified to include the blob gas fee.
+
+```python
+def notify_new_payload(self: ExecutionEngine,
+                       execution_payload: ExecutionPayload,
+                       parent_beacon_block_root: Root,
+                       base_fee_per_blob_gas: uint64) -> bool:
+    """
+    Return ``True`` if and only if ``execution_payload`` is valid with respect to ``self.execution_state``.
+    """
+    ...
+```
+
+##### Modified `verify_and_notify_new_payload`
+
+```python
+def verify_and_notify_new_payload(self: ExecutionEngine,
+                                  state: BeaconState,
+                                  new_payload_request: NewPayloadRequest) -> bool:
+    """
+    Return ``True`` if and only if ``new_payload_request`` is valid with respect to ``self.execution_state``.
+    """
+    execution_payload = new_payload_request.execution_payload
+    parent_beacon_block_root = new_payload_request.parent_beacon_block_root
+
+    if not self.is_valid_block_hash(execution_payload, parent_beacon_block_root):
+        return False
+
+    if not self.is_valid_versioned_hashes(new_payload_request):
+        return False
+
+    # [Modified in Electra]
+    if not self.notify_new_payload(execution_payload,
+                                   parent_beacon_block_root,
+                                   get_base_fee_per_blob_gas(state)):
+        return False
+    
+    state.excess_blob_gas += execution_payload.blob_gas_used
```

Comment:
Something I do not like about this PR is that it forces the CL to treat an UINT256 and do arithmetics with it when the native arithmetics in the CL is on uint64.

Reply:
Actually, all the inputs are `uint64`, so the only place a uint256 is needed is for the output?

Reply:
Ahh my bad, I misunderstood where the arguments where being taken from.  I have no complains indeed.